support-workers/src/typescript/emailFields/paymentDescription.ts

```diff
@@ -0,0 +1,194 @@
+import { partition } from '@modules/arrayFunctions';
+import type { IsoCurrency } from '@modules/internationalisation/currency';
+import { getCurrencyInfo } from '@modules/internationalisation/currency';
+import { isNonEmpty } from '@modules/nullAndUndefined';
+import { BillingPeriod } from '@modules/product/billingPeriod';
+import dayjs from 'dayjs';
+import type { Payment, PaymentSchedule } from '../model/paymentSchedule';
+
+function billingPeriodNoun(period: BillingPeriod): string {
+	switch (period) {
+		case BillingPeriod.Monthly:
+			return 'month';
+		case BillingPeriod.Quarterly:
+			return 'quarter';
+		case BillingPeriod.Annual:
+			return 'year';
+		default:
```

Comment:
Given that there's only one other branch, what do you think about explicitly adding a case for one time rather than using a default? I worry about new values of sum types being added (probably unlikely in this case) and falling silently through to the default which may not be what we want. If there's no default the compiler will complain if a new value is added to the sum type so we'll be forced to think about it.

Reply:
This has actually made me realise that this can be a `RecurringBillingPeriod` not a `BillingPeriod` so we don't need the extra case.